assembly/src/v1/parsers/stack_ops.rs

```diff
@@ -134,9 +134,103 @@ pub fn parse_dupw(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), Assem
     Ok(())
 }
 
-/// TODO: implement
-pub fn parse_swap(_span_ops: &mut Vec<Operation>, _op: &Token) -> Result<(), AssemblyError> {
-    unimplemented!()
+/// Translates swap.x assembly instruction to VM operations MOVUPX MOVDN(X-1)
+pub fn parse_swap(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
+    match op.num_parts() {
+        0 => return Err(AssemblyError::missing_param(op)),
+        1 => span_ops.push(Operation::Swap),
+        2 => match op.parts()[1] {
+            "1" => span_ops.push(Operation::Swap),
+            "2" => {
+                span_ops.push(Operation::MovUp2);
+                span_ops.push(Operation::Swap);
+            }
+            "3" => {
+                span_ops.push(Operation::MovUp3);
+                span_ops.push(Operation::MovDn2);
+            }
```

Comment:
Similar to the other comment, I think this is reversed. It should be: `MOVDN3 MOVUP2`

I think all other similar operations are also reversed.